rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -55,7 +56,15 @@ class Publisher
     intra_process_publisher_handle_(nullptr),
     topic_(topic), queue_size_(queue_size),
     intra_process_publisher_id_(0), store_intra_process_message_(nullptr)
-  {}
+  {
+    // Life time of this object is tied to the publisher handle.
+    if (rmw_get_gid_for_publisher(publisher_handle_, &rmw_gid_) != RMW_RET_OK) {
+      // *INDENT-OFF* (prevent uncrustify from making unecessary indents here)
+      throw std::runtime_error(
+        std::string("failed to create publisher gid: ") + rmw_get_error_string_safe());
```

Comment:
I would suggest `failed to get publisher gid: ` since it does not involve creation / allocation which needs to be deallocated.